### ISSUE TITLE
Test failures

### ISSUE THREAD

**Issue by FRidh** [mpdavis/python-jose#149]
```
============================= test session starts ==============================
platform linux -- Python 3.7.4, pytest-5.1.0, py-1.8.0, pluggy-0.12.0
rootdir: /build/source, inifile: setup.cfg, testpaths: tests
plugins: cov-2.6.1
collected 184 items                                                            

tests/test_firebase.py ...                                               [  1%]
tests/test_jwk.py .........                                              [  6%]
tests/test_jws.py ......................................                 [ 27%]
tests/test_jwt.py .FF..............................................      [ 53%]
tests/test_utils.py ..                                                   [ 54%]
tests/algorithms/test_EC.py .........................                    [ 68%]
tests/algorithms/test_HMAC.py ...                                        [ 70%]
tests/algorithms/test_RSA.py ........................................... [ 93%]
.......                                                                  [ 97%]
tests/algorithms/test_base.py ..                                         [ 98%]
tests/rfc/test_rfc7520.py ...                                            [100%]

=================================== FAILURES ===================================
_______________________ TestJWT.test_invalid_claims_json _______________________

self = <tests.test_jwt.TestJWT object at 0x7ffff5521ad0>

    def test_invalid_claims_json(self):
        old_jws_verify = jws.verify
        try:
            token = u'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhIjoiYiJ9.jiMyrsmD8AoHWeQgmxZ5yq8z0lXS67_QGs52AzC8Ru8'
    
            def return_invalid_json(token, key, algorithms, verify=True):
                return b'["a", "b"}'
    
            jws.verify = return_invalid_json
    
>           with pytest.raises(JWTError, message='Invalid payload string: ["a", "b"}'):
E           TypeError: Unexpected keyword arguments passed to pytest.raises: message
E           Use context-manager form instead?

tests/test_jwt.py:65: TypeError
_________________________ TestJWT.test_invalid_claims __________________________

self = <tests.test_jwt.TestJWT object at 0x7ffff552ed50>

    def test_invalid_claims(self):
        old_jws_verify = jws.verify
        try:
            token = u'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhIjoiYiJ9.jiMyrsmD8AoHWeQgmxZ5yq8z0lXS67_QGs52AzC8Ru8'
    
            def return_encoded_array(token, key, algorithms, verify=True):
                return b'["a","b"]'
    
            jws.verify = return_encoded_array
    
>           with pytest.raises(JWTError, message='Invalid payload string: must be a json object'):
E           TypeError: Unexpected keyword arguments passed to pytest.raises: message
E           Use context-manager form instead?

tests/test_jwt.py:80: TypeError
=============================== warnings summary ===============================
jose/jwt.py:5
  /build/source/jose/jwt.py:5: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Mapping

jose/jws.py:6
  /build/source/jose/jws.py:6: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop working
    from collections import Mapping, Iterable

-- Docs: https://docs.pytest.org/en/latest/warnings.html

----------- coverage: platform linux, python 3.7.4-final-0 -----------
Name                                    Stmts   Miss  Cover   Missing
---------------------------------------------------------------------
jose/__init__.py                            8      0   100%
jose/backends/__init__.py                  11      7    36%   4-8, 12-13
jose/backends/base.py                      13      3    77%   15, 18, 21
jose/backends/cryptography_backend.py     193      6    97%   201, 256, 265-266, 296, 308
jose/backends/ecdsa_backend.py             75      1    99%   88
jose/backends/pycrypto_backend.py         110      6    95%   25, 129-130, 135-136, 152
jose/backends/rsa_backend.py              143      4    97%   73, 128, 190, 210
jose/constants.py                          20      0   100%
jose/exceptions.py                         18      0   100%
jose/jwk.py                                82      4    95%   14-15, 19-20
jose/jws.py                               119      6    95%   156-157, 186-187, 268-269
jose/jwt.py                               154      6    96%   144-145, 148, 435-437
jose/utils.py                              57     21    63%   18-35, 124-134
---------------------------------------------------------------------
TOTAL                                    1003     64    94%

================== 2 failed, 182 passed, 2 warnings in 37.79s ==================
builder for '/nix/store/jxcmvbqgqzrq61272ha8hw52ggn3bwjf-python3.7-python-jose-3.0.1.drv' failed with exit code 1
```

**Comment by blag** [comment]
Please answer ALL of the following questions:

1. How are you running tests?
2. Does this happen with the latest release (as of this writing: 3.1.0)?
3. Does this happen with the `master` branch? And if so, what is the latest commit hash you are testing against?

**Comment by zejn** [comment]
This was fixed in https://github.com/mpdavis/python-jose/commit/c9501b262df9e40fb47cfa1917c58645ec8e4739 and https://github.com/mpdavis/python-jose/commit/aac4c32df210fce3da69e98822e298b3f014fe13

Both fixes are included in release 3.1.0.
